minari/dataset/storages/hdf5_storage.py

```diff
@@ -0,0 +1,217 @@
+from __future__ import annotations
+
+import os
+import pathlib
+from collections import OrderedDict
+from typing import Dict, Iterable, List, Optional, Tuple, Union
+
+import gymnasium as gym
+import h5py
+import numpy as np
+
+from minari.dataset.minari_storage import MinariStorage
+
+
+class _HDF5Storage(MinariStorage):
+    def __init__(
+        self,
+        data_path: pathlib.Path,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+    ):
+        super().__init__(data_path, observation_space, action_space)
+        file_path = os.path.join(self.data_path, "main_data.hdf5")
+        if not os.path.exists(file_path):
+            raise ValueError(f"No data found in data path {self.data_path}")
+        self._file_path = file_path
+
+    @classmethod
+    def _create(
+        cls,
+        data_path: pathlib.Path,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+    ) -> MinariStorage:
+        data_path.joinpath("main_data.hdf5").touch(exist_ok=False)
+        obj = cls(data_path, observation_space, action_space)
+        return obj
+
+    def update_episode_metadata(
```

Comment:
Are we planing to keep using the hdf5 to keep the medata?

Reply:
For the dataset metadata, I would say no. For episode metadata (that now are not accessible btw) I think we can keep it format-specific.